components/gateway/pkg/proxy/transport.go

```diff
@@ -161,10 +179,39 @@ func (t *Transport) getClaims(headers http.Header) (Claims, error) {
 	token = strings.TrimPrefix(token, "Bearer ")
 
 	parser := jwt.Parser{SkipClaimsValidation: true}
-	claims := Claims{}
-	_, _, err := parser.ParseUnverified(token, &claims)
+	tokenClaims := TokenClaims{}
+	_, _, err := parser.ParseUnverified(token, &tokenClaims)
+
+	if err != nil {
+		return Claims{}, errors.Wrap(err, "while parsing bearer token")
+	}
+
+	err = json.Unmarshal([]byte(tokenClaims.ConsumersString), &tokenClaims.Consumers)
 	if err != nil {
-		return claims, errors.Wrap(err, "while parsing beaerer token")
+		return Claims{}, errors.Wrap(err, "while extracting consumers from token")
 	}
+
+	err = json.Unmarshal([]byte(tokenClaims.TenantString), &tokenClaims.Tenant)
+	if err != nil {
+		return Claims{}, errors.Wrap(err, "while extracting tenants from token")
+	}
+
+	claims := Claims{
+		ConsumerTenant: tokenClaims.Tenant["consumerTenant"],
```

Comment:
"consumerTenant" can be exported to a const or if such one form other package exists - it can be reused